ui/src/components/DialogAction/index.jsx

```diff
@@ -122,6 +122,7 @@ export default class DialogAction extends Component {
           </Button>
           <div className={classes.executingActionWrapper}>
             <Button
+              action={actions => actions.focusVisible()}
```

Comment:
Looks good. Let's make the focus visible an opt-in feature to this component as well as allow focus on the secondary action button. I think the following diff would do it:

```diff
diff --git a/ui/src/components/DialogAction/index.jsx b/ui/src/components/DialogAction/index.jsx
index f60b87c5e..baf145e8f 100644
--- a/ui/src/components/DialogAction/index.jsx
+++ b/ui/src/components/DialogAction/index.jsx
@@ -34,6 +34,8 @@ export default class DialogAction extends Component {
     error: null,
     onError: null,
     onComplete: null,
+    focusOnPrimary: false,
+    focusOnSecondary: false,
   };
 
   static propTypes = {
@@ -60,6 +62,14 @@ export default class DialogAction extends Component {
     onClose: func.isRequired,
     /** Error to display. */
     error: oneOfType([string, object]),
+    /** If true, the primary action will have its focus visible
+     * on dialog mount.
+     * */
+    focusOnPrimary: bool,
+    /** If true, the secondary action will have its focus visible
+     * on dialog mount.
+     * */
+    focusOnSecondary: bool,
   };
 
   state = {
@@ -101,6 +111,8 @@ export default class DialogAction extends Component {
       onClose,
       open,
       error,
+      focusOnPrimary,
+      focusOnSecondary,
       onSubmit,
       onComplete,
       ...props
@@ -118,17 +130,19 @@ export default class DialogAction extends Component {
           <DialogContentText>{body}</DialogContentText>
         </DialogContent>
         <DialogActions>
-          <Button disabled={executing} onClick={onClose}>
+          <Button
+            disabled={executing}
+            onClick={onClose}
+            action={actions => focusOnSecondary && actions.focusVisible()}>
             Cancel
           </Button>
           <div className={classes.executingActionWrapper}>
             <Button
-              action={actions => actions.focusVisible()}
+              action={actions => focusOnPrimary && actions.focusVisible()}
               disabled={executing}
               onClick={this.handleSubmit}
               color="secondary"
-              variant="outlined"
-              autoFocus>
+              variant="outlined">
               {confirmText}
             </Button>
             {executing && (
diff --git a/ui/src/components/HookForm/index.jsx b/ui/src/components/HookForm/index.jsx
index 69b93c584..13fc50a05 100644
--- a/ui/src/components/HookForm/index.jsx
+++ b/ui/src/components/HookForm/index.jsx
@@ -752,6 +752,7 @@ export default class HookForm extends Component {
         )}
         {dialogOpen && (
           <DialogAction
+            focusOnPrimary
             fullScreen
             open={dialogOpen}
             onSubmit={this.handleTriggerHookSubmit}
diff --git a/ui/src/views/Tasks/TaskGroup/index.jsx b/ui/src/views/Tasks/TaskGroup/index.jsx
index 617a96a5d..494321ca8 100644
--- a/ui/src/views/Tasks/TaskGroup/index.jsx
+++ b/ui/src/views/Tasks/TaskGroup/index.jsx
@@ -703,6 +703,7 @@ export default class TaskGroup extends Component {
         ) : null}
         {dialogOpen && (
           <DialogAction
+            focusOnPrimary
             fullScreen={Boolean(selectedAction.schema)}
             open={dialogOpen}
             error={dialogError}
diff --git a/ui/src/views/Tasks/ViewTask/index.jsx b/ui/src/views/Tasks/ViewTask/index.jsx
index 7326ff8f7..15021ea87 100644
--- a/ui/src/views/Tasks/ViewTask/index.jsx
+++ b/ui/src/views/Tasks/ViewTask/index.jsx
@@ -859,6 +859,7 @@ export default class ViewTask extends Component {
                   ),
                   confirmText: selectedAction.title,
                 }}
+                focusOnPrimary
                 open={dialogOpen}
                 error={dialogError}
                 onError={this.handleTaskActionError}
```